### ISSUE TITLE
Error when the specified maximum delay is greater than that observed in the data

### ISSUE THREAD

**Issue by seabbs** [epinowcast/epinowcast#167]
Currently when the maximum delay is greater than that observed in the data an error occurs. In reality this should really lead to a warning/message and the maximum delay should be updated to be the maximum observed in the data.

Alternatively when wanting to not use a truncated distribution the default could be to make the code robust to this (though likely with a warning to prevent naive users having very long run-times).

**Comment by seabbs** [comment]
When we construct data that by design does not have sufficient delays for the specified max delay we see that the current default is in fact not to error but to silently the maximum delay present in the data. 


```r
library(epinowcast)
library(data.table)
 
# Filter example hospitalisation data to be natioanl and over all ages
 nat_germany_hosp <- germany_covid19_hosp[location == "DE"]
 nat_germany_hosp <- nat_germany_hosp[age_group %in% "00+"]

filt_dt <- nat_germany_hosp |>
        enw_filter_reference_dates(include_days = 40) |>
        enw_filter_report_dates(remove_days = 20)

enw_preprocess_data(filt_dt, max_delay = 30)
#                   obs          new_confirm              latest
#                <list>               <list>              <list>
#1: <data.table[231x9]> <data.table[231x11]> <data.table[21x10]>
#   missing_reference  reporting_triangle      metareference
#             <list>              <list>             <list>
#1: <data.table[0x6]> <data.table[21x23]> <data.table[21x9]>
#           metareport          metadelay  time snapshots     by groups
#                <list>             <list> <int>     <int> <list>  <int>
#1: <data.table[50x12]> <data.table[30x4]>    21        21             1
#   max_delay   max_date
#       <num>     <IDat>
#1:        30 2021-09-30
```

**Comment by seabbs** [comment]
Fixed in #190.
